### PR TITLE
Add test_collect_temp_data()

### PR DESCRIPTION
**Pull request by anschweitzer**
Adds a simple test that verifies PrimaryScada receives publication from TankWaterTempSensor.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -23,3 +23,5 @@ jobs:
           pip install -r gw_spaceheat/requirements/test.txt
       - name: Run tests with pytest
         run: pytest
+        env:
+          PYTHONPATH: gw_spaceheat:$PYTHONPATH
```
Comment (jessicamillar) [inline .github/workflows/ci.yaml:27]: Nice. We could use one of the online brokers for the github tests, so if we figure out how to copy files over in git actions then maybe ci will start to work.

File: test/test_examples.py [hunk 2]
```diff
@@ -1,17 +1,40 @@
...
+        self.num_received_by_topic = defaultdict(int)
+        super().__init__(node)
+
+    def on_mqtt_message(self, client, userdata, message):
+        self.num_received += 1
+        self.num_received_by_topic[message.topic] += 1
+        super().on_mqtt_message(client, userdata, message)
+
+    def gs_dispatch_received(self, payload: GsDispatch, from_node: ShNode):
+        raise NotImplementedError
```
Comment (jessicamillar) [inline test/test_examples.py:37]: I've cleaned up the `atn_base.py` and `primary_scada_base.py`, so abstractmethod `gs_dispatch_received` is gone and `on_gw_message` exists. See [https://github.com/thegridelectric/gw-scada-spaceheat-python/blob/jm/async-power-test/gw_spaceheat/actors/primary_scada_base.py#L44] here in the other PR

File: test/test_examples.py [hunk 3]
```diff
@@ -75,3 +98,17 @@ def test_async_power_metering_dag():
...
+    load_house.load_all(house_json_file='../test/test_data/test_load_house.json')
+    scada = ScadaRecorder(node=typing.cast(ShNode, ShNode.by_alias["a.s"]))
+    thermo = TankWaterTempSensor(node=typing.cast(ShNode, ShNode.by_alias["a.tank.temp0"]))
+    time.sleep(1)
+    thermo.terminate_sensing()
+    thermo.sensing_thread.join()
+    scada.terminate_scheduling()
+    scada.schedule_thread.join()
+    assert scada.num_received > 0
+    assert scada.num_received_by_topic["a.tank.temp0/gt.telemetry.110"] == scada.num_received
```
Comment (jessicamillar) [inline test/test_examples.py:114]: When we update the versioning of the schema, the topic will change and this line will have to change. But that seems like a good thing. Plus it is helpful to actually see what the topic looks like.

On another note, I was wondering why the tests terminated without calling `scada.gw_consume_client.loop_stop()` (etc for all 4 mqtt client loops for scada, and both mqtt client loops for thermo)

**Review by jessicamillar — APPROVED** [review]
Looks good to merge

File: test/test_examples.py [hunk 4]
```diff
@@ -75,3 +98,17 @@ def test_async_power_metering_dag():
     meter.publish(payload=payload)
     time.sleep(.3)
 #     assert atn.total_power_w == 2100
+
+
+def test_collect_temp_data():
+    """Verify Scada receives publication from TankWaterTempSensor"""
+    load_house.load_all(house_json_file='../test/test_data/test_load_house.json')
+    scada = ScadaRecorder(node=typing.cast(ShNode, ShNode.by_alias["a.s"]))
+    thermo = TankWaterTempSensor(node=typing.cast(ShNode, ShNode.by_alias["a.tank.temp0"]))
```
Comment (jessicamillar) [inline test/test_examples.py:107]: Interesting

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (anschweitzer) [reply .github/workflows/ci.yaml:27]: There's probably an open source recipe to start an mqtt broker, probably mosquitto in a docker container. I just need to do a little GitHub Actions reading before pushing that further.

**Pull request merged**